### ISSUE TITLE
Documentation on usage in linux

### ISSUE THREAD

**Issue by kkalera** [tangrams/tangram-es#767]
Hi everyone,

I'm looking for some information on how to use the tangram map. I am running it inside Ubuntu. I am also running the valhalla server and would like to use them both for navigational purposes. The valhalla package is working fine for me. Its just that i'm struggling a bit with Tangram-es. If I understand correctly, in the JS version of Tangram, we use leaflet to draw lines and communicate with the server.

However, I can't seem to find any documentation on how to draw for instance a polyline on the map in Tangram-es. I don't need an extremely detailed explanation of how Tangram-es works. Or am I asking to write me an entire guide.

If anyone could just point me in the right direction, I can probably figure it out. I am more than willing to document my steps in the wiki so that others can learn from this as well.

Thank you in advance.

**Comment by blair1618** [comment]
Hello! The tangram-es interface for Linux and OS X are definitely a little rough around the edges, we don't have a supported "SDK" for those the way we do for Android. That said, you can use basically all of tangram's features on any of our build targets. 

TL;DR - check out the OS X demo app code to see how to add points and polylines: https://github.com/tangrams/tangram-es/blob/master/osx/src/main.cpp

The way that these dynamically-created geometries work is clunky right now and will be changing soon, but essentially you add and remove dynamic geometry through a `DataSource` object, specifically a `ClientGeoJsonSource`. You create a `ClientGeoJsonSource`, add it to tangram with `Tangram::addDataSource()`, add points, polylines, polygons, or a string of GeoJSON to it, and then those features will appear on the map. You use the scene file to style your dynamic features the same way that you style bas map features, meaning that for your features to show up at all there must be styling rules for them. 

Hope that helps, let me know if you have more specific questions.

**Comment by kkalera** [comment]
Hello @blair1618 ,thank you for the information. I was also wondering what the easiest way is, to give some inputs to tangram es. Let's say for instance, make the map go to a certain lat-lng location. Is this possible via a terminal window? If not, what would be the best way to implement this?

Thank you for all the help.

**Comment by blair1618** [comment]
There's nothing built into tangram-es that would enable text input from a command line (it's difficult to generalize such a thing across desktop and mobile operating systems). However in a desktop application, you can pass locations as command line arguments (like we do for a scene file argument [here](https://github.com/tangrams/tangram-es/blob/master/osx/src/main.cpp#L333-L340) or use the standard input streams to easily accept and parse text input from the command line.

**Comment by blair1618** [comment]
Adding to my previous comment: If you're imagining some kind of graphical interface for setting location, that would be something you need to add yourself, though there are nice libraries for doing this in C++ with opengl (https://github.com/ocornut/imgui). GLFW also has simple ways to receive text input if you want something more basic (http://www.glfw.org/docs/latest/input_guide.html#input_char).

**Comment by kkalera** [comment]
Hi @blair1618 , that was pretty much the information I was looking for. Thank you.
